src/components/ContentPreview/ContentPreview.js

```diff
@@ -598,12 +613,16 @@ class ContentPreview extends PureComponent<Props, State> {
         if (!this.isPreviewLibraryLoaded() || !file || !tokenOrTokenFunction) {
             return;
         }
-
-        const typedId: string = getTypedFileId(this.getFileId(file));
+        const fileId = this.getFileId(file);
+        const typedId: string = getTypedFileId(fileId);
         const token: TokenLiteral = await TokenService.getReadToken(
             typedId,
             tokenOrTokenFunction,
         );
+        if (fileId !== this.state.currentFileId) {
```

Comment:
State is destructured above.
Should this check be moved above, so that token call/XHR may be prevented?

Reply:
Yes, it is destructured above, but ```currentFileId``` cannot move there. This code is async, so the value of the state could change during this time and the value of ```currentFileId``` would be stale

Reply:
Moving it above is also not something which has to be done, as we definitely want it in after the await. Adding an additonal check before is not needed, as the function wouldnt be called in the first place if they were different (see ```componentDidUpdate```)